Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -480,12 +480,12 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
     }
 
     // http://webkit.org/b/190138 Footers should become contentInfo's if scoped to body (and consequently become a landmark).
-    // It should remain a footer if scoped to main, sectioning elements (article, aside, nav, section) or root sectioning element (blockquote, details, dialog, fieldset, figure, td).
+    // It should remain a footer if scoped to main, sectioning elements (article, aside, nav, section).
     // https://w3c.github.io/html-aam/#el-footer
     if (node->hasTagName(footerTag)) {
-        if (!isDescendantOfElementType({ articleTag, asideTag, navTag, sectionTag, mainTag, blockquoteTag, detailsTag, dialogTag, fieldsetTag, figureTag, tdTag }))
+        if (!isDescendantOfElementType({ articleTag, asideTag, mainTag, navTag, sectionTag }))
             return AccessibilityRole::LandmarkContentInfo;
-        return AccessibilityRole::Footer;
+        return AccessibilityRole::Generic;
```

Comment:
This removes the only place `AccessibilityRole::Footer` is ever assigned, so if we go this route we'd probably just want to remove the role entirely. And I don't think that's the right move, as we check for this role in several functions, so removing it would have negative ramifications (i.e. the usage in `WebAccessibilityObjectWrapperIOS`).

Reply:
According to the specification, it should return a generic role like the header element. We need to find another way to return the footer role for this case.